docs: Add a section about Expo Modules API (#4291)
facebook/react-native-website: docs/native-platforms.md

   - [Advanced: Custom C++ Types](the-new-architecture/custom-cxx-types.md)
2. Fabric Native Components
   - [Android & iOS](fabric-native-components.md)

## Native development with Expo

> Expo provides **Expo Modules API** to write platform and renderer agnostic native modules and views. The API takes advantage of modern native language features from Swift and Kotlin, has built-in type conversion and safety, and provides a similar developer experience between each platform, so it's effortless to jump between Android and iOS modules. It also has similar performance characteristics to Turbo Modules API because it's built with React Native's JavaScript Interface (JSI).

Reviewer: ```suggestion
The **Expo Modules API** takes advantage of modern native language features from Swift and Kotlin, has built-in type conversion and safety, and provides a similar developer experience on each platform, so it's effortless to jump between Android and iOS modules. It has similar performance characteristics to Turbo Modules API because it's built with React Native's JavaScript Interface (JSI).
```
